src/silicone/database_crunchers/equal_quantile_walk.py

```diff
@@ -0,0 +1,157 @@
+                        lead_unit, lead_in.variables(True)["unit"].tolist()
+                    )
+                )
+
+            if data_follower_time_col != in_iamdf.time_col:
+                raise ValueError(
+                    "`in_iamdf` time column must be the same as the time column used "
+                    "to generate this filler function (`{}`)".format(
+                        data_follower_time_col
+                    )
+                )
+            if lead_in.data.empty:
+                raise ValueError(
+                    "There is no data for {} so it cannot be infilled".format(
+                        variable_leaders
+                    )
+                )
+            output_ts = lead_in.timeseries()
+            if any(
+                [
+                    (time not in lead_ts.columns) or (time not in follower_ts.columns)
+                    for time in output_ts.columns
+                ]
+            ):
+                # We allow for cases where either lead or follow have gaps
+                raise ValueError(
+                    "Not all required timepoints are present in the database we "
+                    "crunched, we crunched \n\t{} for the lead and \n\t{} for the "
+                    "follow \nbut you passed in \n\t{}".format(
+                        lead_ts.columns, follower_ts.columns, output_ts.columns
+                    )
+                )
+            for col in output_ts.columns:
+                output_ts[col] = self._find_same_quantile(
+                    follower_ts[col], lead_ts[col], output_ts[col]
+                )
+            output_ts = output_ts.reset_index()
+            output_ts["variable"] = variable_follower
+            output_ts["unit"] = data_follower_unit
+            return IamDataFrame(output_ts)
+
+        return filler
+
+    def _get_iamdf_follower(self, variable_follower, variable_leaders):
+        if len(variable_leaders) > 1:
+            raise ValueError(
+                "For `EqualQuantileWalk`, ``variable_leaders`` should only "
+                "contain one variable"
+            )
+
+        self._check_follower_and_leader_in_db(variable_follower, variable_leaders)
+
+        return self._db.filter(variable=variable_follower)
+
+    def _find_same_quantile(self, follow_vals, lead_vals, lead_input):
+        if len(lead_vals) == 1:
+            # If there is only a single value we have to return that.
+            return follow_vals[0]
+        lead_vals = lead_vals.sort_values()
+        quant_of_lead_vals = np.arange(len(lead_vals)) / (len(lead_vals) - 1)
+        if any(quant_of_lead_vals > 1) or any(quant_of_lead_vals < 0):
+            raise NotImplementedError("Impossible quantiles!")
+        input_quantiles = scipy.interpolate.interp1d(
+            lead_vals, quant_of_lead_vals, bounds_error=False, fill_value=(0, 1)
+        )(lead_input)
+        return np.nanquantile(follow_vals, input_quantiles)
```

Comment:
```suggestion
        return np.nanquantile(follow_vals, input_quantiles, interpolation="linear")
```

It's already what happens but just makes clear?